pkg/dao/rpm.go

```diff
@@ -0,0 +1,171 @@
+package dao
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"gorm.io/gorm"
+)
+
+type rpmDaoImpl struct {
+	db *gorm.DB
+}
+
+func GetRpmDao(db *gorm.DB) RpmDao {
+	return rpmDaoImpl{
+		db: db,
+	}
+}
+
+func (r rpmDaoImpl) isOwnedRepository(orgID string, accountID string, repoUUID string) error {
+	var repoConfigs []models.RepositoryConfiguration
+	if err := r.db.
+		Where("org_id = ? and account_id = ? and repository_uuid = ?", orgID, accountID, repoUUID).
+		Find(&repoConfigs).
+		Error; err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r rpmDaoImpl) Create(orgID string, accountID string, repo *models.Repository, newRpm *models.Rpm) error {
+	// Check arguments
+	if orgID == "" {
+		return fmt.Errorf("orgID can not be an empty string")
+	}
+	if accountID == "" {
+		return fmt.Errorf("accountID can not be an empty string")
+	}
+	if repo == nil {
+		return fmt.Errorf("repo can not be nil")
+	}
+	if newRpm == nil {
+		return fmt.Errorf("newRpm can not be nil")
+	}
```

Comment:
Moving the validations to the model here would be preferred too

Reply:
sorry but I cannot move this validations to the model; they are validating nil references from the arguments but not the content of the fields.

Reply:
oh right!  i misread